Container/Container.cs

```diff
@@ -1,57 +1,33 @@
-﻿using System.ComponentModel.DataAnnotations;
-
-namespace Container
+﻿namespace Container
 {
     public class Container
     {
-        List<int> container = new List<int>();
-        public int maxLength = 5;
+        private readonly List<int> _container = new(capacity: 5);
+
+        public int MaxLength { get; } = 5;
 
-        public void TryAdd(int value)
+        public bool TryAdd(int value)
         {
-            if (container.Count < maxLength+1)
-            {
-                container.Add(value);
-            }
-            else
+            if (_container.Count == MaxLength)
```

Comment:
To jest "guard" tej metody. Generalnie jeśli w Twojej metodzie są pewne sytuacje gdzie nie chcesz wykonać jakiejś operacji (lub rzucić wyjątek), rób to na początku. Dopier kiedy przejdziesz przez wszystkie takie "guardy", wykonaj operację, którą powinna zrobić metoda.